### PR TITLE
Strip question two down so that a lot of the boilerplate is removed.

### PR DESCRIPTION
**Pull request by brandonwinch**
A lot of our feedback around question two of our technical test revolves the time it takes to get the services connected and then understanding the attached Swimlane components.

In reality, this question is not supposed to be testing their ability to wire up a service injected through dependencies, nor is it to feed data into a swimlane. It's just about data manipulation and combining everything to get a result.

What I've done here is fill out the service so that our candidates know what they should be fetching for and what comes back via Types. Then instead of doing anything with a Swimlane, dump everything out into a JSON blob. Provided that the blob is correct, a unit test will pass.

This should shave a reasonable amount of time off just when it comes to boilerplate work and understanding surrounding components.

### CODE REVIEW

File: src/question-two/QuestionTwo.tsx [hunk 1]
```diff
@@ -4,13 +4,17 @@ import { IAppTabContainer } from "../common/types"
...
+  title: string
+  cards: {
+    allocId: number,
+    allocType: 'job' | 'activity',
+    resourceId: number,
+    description: string,
+    start: string,
+    end: string
+  }[]
+}
```
Comment (brandonwinch) [inline src/question-two/QuestionTwo.tsx:17]: Should we still call this a `Lane` with cards inside? At first I thought it could probably be a bit more ambiguous, but in the end I guess it doesn't matter _what_ we call it. So long as the shape and data matches.

File: src/question-two/INSTRUCTIONS.md [hunk 2]
```diff
@@ -1,17 +1,8 @@
...
-
-Once you have that data we’d like you to display it in the Swimlane component provided.  Please do not change the shape of the properties being sent to the Swimlane component, you will need to modify the data received from the server to fit the component's structure.
-
-Note: While it would be better to have the server expose an endpoint that provides all this data in a nice format, we want to see what you do when given data from a couple of different sources.
-
-Note: We have provided a `service` prop to the component that is defined in /src/service/DataService.js.  It has access to a graphql client and a http request library.  If you wish to fetch data from the server we suggest you do it from there.
+For this question we want you to fetch data from a couple of endpoints on the server.  This data will describe the events of some people’s day (specifically 01/09/2018).  We’d like you to take that data, merge it together and render it as a JSON blob in the component.
 
+There is a unit test that contains an expected output from the component. Getting this test to pass is the goal.
```
Comment (lisatassone) [inline src/question-two/INSTRUCTIONS.md:5]: `the expected output`?

File: src/question-two/INSTRUCTIONS.md [hunk 3]
```diff
@@ -1,17 +1,8 @@
...
 ### What we're looking for
 
-The goal of Question 2 is to see how our candidates reason the merging of multiple endpoints and produce a clean code solution. Our primary view will be on code readability and any performance compromises made. Including some testing would be beneficial as well as it would create an opportunity to iterate through the solution while trying to find a solution that is clear.
\ No newline at end of file
+The goal of Question 2 is to see how our candidates reason the merging of multiple endpoints and produce a clean code solution. Our primary view will be on code readability and any performance compromises made. Including some extra tests would be beneficial as well as it would create an opportunity to iterate through the solution while trying to find a solution that is clear.
```
Comment (lisatassone) [inline src/question-two/INSTRUCTIONS.md:8]: You just mentioned the goal as getting the test to pass. How bout we reword this to say:

`We would like to see how our candidates...` `Our primary focus will be...`

File: src/question-two/QuestionTwo.test.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  getActivityAllocations: () => Promise.resolve(data.activityAllocations),
+  getResources: () => Promise.resolve(data.resources)
+}
+
+test('fetched and merged data is in the expected shape', async () => {
+  render(<QuestionTwo service={mockDataService} />)
+
+  await wait()
+
+  expect(screen.getByText('[{"title":"Sam Seaborn","cards":[{"allocId":1,"allocType":"job","resourceId":0,"description":"Shield some wiring","start":"2018-09-01T09:00:00Z","end":"2018-09-01T13:00:00Z"}]},{"title":"Donna Moss","cards":[{"allocId":0,"allocType":"job","resourceId":1,"description":"Build a shed","start":"2018-09-01T10:15:00Z","end":"2018-09-01T11:00:00Z"},{"allocId":0,"allocType":"activity","resourceId":1,"description":"Meal Break","start":"2018-09-01T12:15:00Z","end":"2018-09-01T13:10:00Z"}]},{"title":"Toby Ziegler","cards":[{"allocId":1,"allocType":"activity","resourceId":2,"description":"Meal Break","start":"2018-09-01T12:15:00Z","end":"2018-09-01T13:10:00Z"}]}]'))
```
Comment (lisatassone) [inline src/question-two/QuestionTwo.test.js:20]: Instead of using `await wait()` and `getByText` I think you can use `await screen.findByText()`

File: src/question-two/QuestionTwo.tsx (hunk 1 above)
Comment (lisatassone) [reply src/question-two/QuestionTwo.tsx:17]: `Lane` doesn't really mean anything now. We are wanting people to merge the schedule data of a resource, so how 'bout `ResourceSchedule`? Likewise, some of the keys are off. Maybe instead of `title` it's called `name` and `resourceId` sits outside of cards?

File: src/server/db.json [hunk 5]
```diff
@@ -3,31 +3,31 @@
...
+      "contactId": 0,
       "start": "2018-09-01T09:00:00Z",
       "end": "2018-09-01T13:00:00Z",
       "location": "Brisbane"
     },
     {
       "id": 3,
       "name": "Pick up a trailer",
-      "contactId": "0",
+      "contactId": 0,
```
Comment (lisatassone) [inline src/server/db.json:30]: We have a bug that I've been meaning to fix in here with contacts. Two of them have the same id. Be good to fix that up

**Review by lisatassone — APPROVED** [review]
Some minor suggestions but looks good. Should help out future candidates :)

File: src/server/db.json (hunk 5 above)
Comment (brandonwinch) [reply src/server/db.json:30]: Good catch.

File: src/question-two/QuestionTwo.test.js (hunk 4 above)
Comment (brandonwinch) [reply src/question-two/QuestionTwo.test.js]: *chefkiss*
